### PR TITLE
Swagger for Azure Event Grid.

### PR DESCRIPTION
**Pull request by kalyanaj**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have been fixed in this PR except 1 warning and 1 error, x-ms-examples will be added before 09/15.

### CODE REVIEW

**Comment by salameer** [comment]
@dsgouda  the changes from what we have in Private are straight forward. let's get those guys rockin :)

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-06-15-preview/EventGrid.json [hunk 1]
```diff
@@ -0,0 +1,1760 @@
...
+                    "type": "string"
+                }
+            }
+        },
+        "TopicProperties": {
+            "description": "Properties of the Topic",
+            "type": "object",
+            "properties": {
+                "provisioningState": {
+                    "description": "Provisioning state of the t
```
Comment (dsgouda) [inline specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-06-15-preview/EventGrid.json]: Looks like this really has limited values, consider modelling this as enum

File: specification/eventgrid/resource-manager/readme.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+## Configuration
+
+
+
+### Basic Information
+These are the global settings for the Azure EventGrid API.
+
+``` yaml
+openapi-type: arm
+tag: package-2017-06-preview
```
Comment (dsgouda) [inline specification/eventgrid/resource-manager/readme.md]: nit: we should probably keep tags consistent with the name of the directory in which the json is placed.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Not a lot has changed since the last review. Just a couple of minor suggestions.

File: specification/eventgrid/resource-manager/readme.md (hunk 2 above)
Comment (kalyanaj) [inline specification/eventgrid/resource-manager/readme.md]: Looking at the tag field in a few other services, it seems to follow the pattern of "package-<year>-<month>" or "package-<year>-<month>-preview", the current name (package-2017-06-preview) was chosen to be consistent with that naming scheme, so if it doesn't cause any side-effects, would like to continue with the current name.

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-06-15-preview/EventGrid.json (hunk 1 above)
Comment (kalyanaj) [inline specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-06-15-preview/EventGrid.json]: Sorry I didn't fully get this comment - are you referring to the provisioningState property (which is already an enum), or are you referring to TopicProperties as a whole?

Comment (dsgouda) [inline specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-06-15-preview/EventGrid.json]: I guess github scrambled my comment earlier, I was suggesting making `keyName` an enum.

Comment (kalyanaj) [inline specification/eventgrid/resource-manager/Microsoft.EventGrid/2017-06-15-preview/EventGrid.json]: Got it, thanks. We will consider this change during a future update.

**Review by dsgouda — APPROVED** [review]
Approving per offline discussion

**Comment by dsgouda** [comment]
Overriding CI since there are build issues today and this needs to go in ASAP.

**Pull request merged**